data/net/minecraft/world/item/enchantment/Enchantment.mapping

```diff
@@ -1,55 +1,103 @@
+		COMMENT @return Whether or not the two enchantments are compatible.
+		ARG 1 other
+			COMMENT The other enchantment to test compatibility with.
 	METHOD doPostAttack (Lnet/minecraft/world/entity/LivingEntity;Lnet/minecraft/world/entity/Entity;I)V
-		COMMENT Called whenever a mob is damaged with an item that has this enchantment on it.
-		ARG 1 user
+		COMMENT A hook that is called every time an entity with this enchantment attacks another entity.
+		ARG 1 attacker
+			COMMENT The user of the enchantment.
 		ARG 2 target
+			COMMENT The entity being attacked.
 		ARG 3 level
+			COMMENT The level of the enchantment.
 	METHOD doPostHurt (Lnet/minecraft/world/entity/LivingEntity;Lnet/minecraft/world/entity/Entity;I)V
-		COMMENT Whenever an entity that has this enchantment on one of its associated items is damaged this method will be called.
+		COMMENT A hook that is called every time an entity with this enchantment is attacked by another entity.
 		ARG 1 user
+			COMMENT The user of the enchantment.
 		ARG 2 attacker
+			COMMENT The entity that attacked the user.
 		ARG 3 level
+			COMMENT The level of the enchantment.
 	METHOD getDamageBonus (ILnet/minecraft/world/entity/MobType;)F
-		COMMENT Calculates the additional damage that will be dealt by an item with this enchantment. This alternative to calcModifierDamage is sensitive to the targets EnumCreatureAttribute.
+		COMMENT Calculates the amount of additional damage to deal when attacking with an item with the enchantment.
 		ARG 1 level
-		ARG 2 creatureType
+			COMMENT The level of the enchantment being used.
+		ARG 2 mobType
+			COMMENT The type of mob being damaged.
 	METHOD getDamageProtection (ILnet/minecraft/world/damagesource/DamageSource;)I
-		COMMENT Calculates the damage protection of the enchantment based on level and damage source passed.
+		COMMENT Calculates the amount of additional damage protection to provide when a user is damaged.
 		ARG 1 level
+			COMMENT The level of the enchantment being used.
 		ARG 2 source
+			COMMENT The source of the damage.
 	METHOD getDescriptionId ()Ljava/lang/String;
-		COMMENT Return the name of key in translation table of this enchantment.
+		COMMENT Gets the description Id of the enchantment. This is commonly used to create localization keys.
 	METHOD getFullname (I)Lnet/minecraft/network/chat/Component;
+		COMMENT Gets the name of the enchantment with the level appended to the end. The game will attempt to convert the letter to roman numerals.
+		COMMENT @return The name of the enchantment with the level appened to the end.
 		ARG 1 level
+			COMMENT The level of the enchantment.
 	METHOD getMaxCost (I)I
-		ARG 1 enchantmentLevel
+		ARG 1 level
 	METHOD getMaxLevel ()I
-		COMMENT Returns the maximum level that the enchantment can have.
+		COMMENT Gets the maximum level of the enchantment under normal circumstances.
```

Comment:
What does "under normal circumstances" mean? What are the exceptions?

Reply:
The "normal circumstances" was just meant to suggest that editing NBT, or mods like apotheosis could artificially bypass the limit as they're not strictly enforced. Clarified that a bit.